evm_loader/program/src/executor/precompile_extension/spl_token.rs

```diff
@@ -559,31 +543,21 @@ fn mint<B: AccountStorage>(
 fn burn<B: AccountStorage>(
     context: &crate::evm::Context,
     state: &mut ExecutorState<B>,
+    mint: Pubkey,
     source: Pubkey,
     amount: u64,
 ) -> Result<Vec<u8>> {
     let signer = context.caller;
     let (signer_pubkey, bump_seed) = state.backend.solana_address(&signer);
 
-    let account = state.external_account(source)?;
-    spl_token::check_program_account(&account.owner)?;
-
-    let token_account = spl_token::state::Account::unpack(&account.data)?;
-
     let seeds = vec![
         vec![ACCOUNT_SEED_VERSION],
         signer.as_bytes().to_vec(),
         vec![bump_seed],
     ];
 
-    let burn = spl_token::instruction::burn(
-        &spl_token::ID,
-        &source,
-        &token_account.mint,
-        &signer_pubkey,
-        &[],
-        amount,
-    )?;
+    let burn =
+        spl_token::instruction::burn(&spl_token::ID, &source, &mint, &signer_pubkey, &[], amount)?;
```

Comment:
This code formatting is ugly. Is there any way to fit this into 1 line?

Reply:
Fixed